openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiEndpointMapper.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.ballerina.service;
+
+import io.ballerina.compiler.syntax.tree.ExpressionNode;
+import io.ballerina.compiler.syntax.tree.FunctionArgumentNode;
+import io.ballerina.compiler.syntax.tree.ImplicitNewExpressionNode;
+import io.ballerina.compiler.syntax.tree.ListenerDeclarationNode;
+import io.ballerina.compiler.syntax.tree.MappingConstructorExpressionNode;
+import io.ballerina.compiler.syntax.tree.MappingFieldNode;
+import io.ballerina.compiler.syntax.tree.NamedArgumentNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.ParenthesizedArgList;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SpecificFieldNode;
+import io.swagger.models.Scheme;
+import io.swagger.models.Swagger;
+import org.ballerinalang.ballerina.Constants;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Extract OpenApi server information from and Ballerina endpoint.
+ */
+public class OpenApiEndpointMapper {
+
+    /**
+     * Convert endpoints bound to {@code service} openapi server information.
+     * Currently this method only selects last endpoint bound to {@code service}
+     * as the Server for openapi definition. This is due to OAS2 not supporting
+     * multiple Server definitions.
+     *
+     * @param endpoints all endpoints defined in ballerina source
+     * @param service   service node with bound endpoints
+     * @param openapi   openapi definition to attach extracted information
+     * @return openapi definition with Server information
+     */
+    public Swagger convertBoundEndpointsToOpenApi(List<ListenerDeclarationNode> endpoints,
+                                                  ServiceDeclarationNode service, Swagger openapi) {
+
+//        if (endpoints == null || service == null || service.getAttachedExprs().isEmpty()
+//                || (service.getAttachedExprs().get(0).getKind() != NodeKind.SIMPLE_VARIABLE_REF
+//                && service.getAttachedExprs().get(0).getKind() != NodeKind.TYPE_INIT_EXPR)) {
+//            return openapi;
+//        }
```

Comment:
Is there any reason to keep this commented block? (If so, lets add a `Todo` comment with the reason)

Reply:
yeah , still this code is refactoring with new changes, for the moment commented some code part for having the immediate fix for some blockers. :)